axon-cosmosdb/src/test/java/org/axonframework/extensions/cosmosdb/eventsourcing/tokenstore/CosmosTokenStoreIntegrationTest.java

```diff
@@ -0,0 +1,508 @@
+package org.axonframework.extensions.cosmosdb.eventsourcing.tokenstore;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.thoughtworks.xstream.XStream;
+import org.axonframework.common.AxonConfigurationException;
+import org.axonframework.eventhandling.GlobalSequenceTrackingToken;
+import org.axonframework.eventhandling.Segment;
+import org.axonframework.eventhandling.TrackingToken;
+import org.axonframework.eventhandling.tokenstore.TokenStore;
+import org.axonframework.eventhandling.tokenstore.UnableToClaimTokenException;
+import org.axonframework.eventhandling.tokenstore.UnableToInitializeTokenException;
+import org.axonframework.serialization.Serializer;
+import org.axonframework.serialization.json.JacksonSerializer;
+import org.axonframework.serialization.xml.XStreamSerializer;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.io.*;
+import org.testcontainers.containers.CosmosDBEmulatorContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Testcontainers
```

Comment:
The fact the test container doesn't work on GitHub, makes it kinda sh*tty if **this** is the only test of the `CosmosTokenStore`. So, why didn't you make a unit test too?

Reply:
What would you like to test whithout having a client, not even a mocked client?

Reply:
When dropping the comment, I wasn't aware you couldn't make a mock out of the client... Honestly, I think it's a misser with CosmosDB.
So, if I'd be very picky? An issue with the CosmosDB repository asking for a unit-testable way of CosmosDB for integrators like us.